### PR TITLE
Fixes #1534, added `legacy-peer-deps` option set to `true`

### LINKED ISSUE #1534: Node Version 15+ Does Not Install Dependecies Properly
**What happened**: I have tried to set up the project to run on my local machine. I followed [the environment setup documentation ](https://github.com/Seneca-CDOT/telescope/blob/master/docs/environment-setup.md)and on [the final step](https://github.com/Seneca-CDOT/telescope/blob/master/docs/environment-setup.md#finally) I could not start the backend. `npm start` returned an error:
image
After trying to run `npm install` from the root of the project explicitly, I have received the following error:
image
`node_modules` was not installed in the root directory.

**What should have happened**: Required dependencies should have been installed successfully and the backend should have started.

**How to reproduce it (as precise as possible)**: Install Node version 15+ (specifically mine was `15.5.1`) and follow [the environment setup documentation](https://github.com/Seneca-CDOT/telescope/blob/master/docs/environment-setup.md) for running the project on local. 

**Anything else we need to know?**: I have done some research on the error I got and found [a potentially linked issue open under Jest repository](https://github.com/facebook/jest/issues/10820). As suggested in Telescope's Slack channel, we might either try to manually fix this dependencies issue or set explicitly the Node engine to use version 14 until the newer version has this issue fixed.

**Environment**: Production, `Node` version `15.5.1`, `npm` version `7.3.0`

- OS: Ubuntu Linux 20.04.1 LTS (Potentially Windows and macOS as well)
- Browser: N/A

### PR DESCRIPTION
**Pull request by Metropass**
## Type of Change
Fixes #1534 

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
As of npm7.x, Peer Dependencies has had an overhaul, causing some of our dependencies not properly installing if a user runs `npm install`. This is a band-aid fix to use the legacy Peer Dependencies, until we fix the issue for forwards compatibility. 

More Information on the Changes of Peer Dependencies can be found [here](https://github.com/npm/rfcs/blob/latest/implemented/0025-install-peer-deps.md)  

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by Metropass** [comment]
Saved as a draft for now, due to my other PR not being merged yet.

File: .npmrc [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 package-lock=false
+
 # Use exact versions of dependencies vs. semver minor updates
 save-exact=true
 
 # Allow running postinstall script in Docker
 unsafe-perm=true
+
+# Allows users of npm7.x to properly build Telescope without breaking. Uses old npm6.x peer dependency installation
+legacy-peer-deps=true
```
Comment (humphd) [inline .npmrc:11]: Add a newline after this

**Review by humphd — CHANGES_REQUESTED** [review]
Can you rebase this on master, and drop the other changes in here?  Then we could land it separate to your FRONTEND fix.

File: .npmrc (hunk 1 above)
Comment (Metropass) [reply .npmrc:11]: Will do! I'll also rebase to master so that we can clean things up a bit.

**Review by humphd — CHANGES_REQUESTED** [review]
This still isn't right @Metropass, can you please rebase this properly so you only have 1 commit and have it be off of `master`?

If you need help, please talk with folks on slack

**Comment by Metropass** [comment]
Closed, as this PR is very outdated. See #1694 as it's a cleaner version and rebased

**Pull request closed without merging**